player.js

```diff
@@ -4,11 +4,13 @@ class Player {
   }
 
   tagPlayer (predator) {
-    if (predator.tag) {
-      this.tag = true
-    } else if (!predator.tag && this.tag) {
-      predator.tag = true
-    }
+    if (predator.tag) this.tag = true
+    else if (this.tag) predator.tag = true
+  }
+
+  static validateName (name) {
+    if (typeof name !== 'string') return false
+    return /^[0-9A-Za-z ]{1,15}$/.test(name) && name.replace(/\s/g, '') !== ''
```

Comment:
Okay, this is old code, but I'd find it preferable if this was changed, since the RegExp object gets recreated every time the function gets run, which could lead to performance issues. While it's not run as frequently, it'd not hurt to do it.

Reply:
Good point, I'll fix that!